### DISCUSSION TITLE
Unable to convert key: num when using 'partition_on' argument

### DISCUSSION THREAD

**Discussion by austinvdubs** [sfu-db/connector-x#358] [Q&A]
Hello, I was having an issue with adding the 'partition_on' argument to the query. I outlined it more clearly on Stack Overflow here:
https://stackoverflow.com/questions/73792804/connectorx-unable-to-convert-key-num-when-using-partition-on-argument
(I don't have enough reputation on Stack Overflow to create a new tag for `connectorx`, so I'm cross posting it here for visibility)

I also submitted another question on StackOverflow about a potential bug in handling cx.read_sql() on a database table that is being concurrently updated (https://stackoverflow.com/questions/73794989/connectorx-cx-read-sql-returns-dataframe-of-all-zeros-rows-intermittently-or)

**Comment by wangxiaoying** [comment]
Hi @austinvdubs , it looks like the error is caused by partitioning the query on the `quoteTimeInLong` column. May I ask what's the data type of `quoteTimeInLong`? It will be helpful to reproduce the error locally.

**Reply by austinvdubs** [reply]
Hello! Thanks for getting back to me. `quoteTimeInLong` is an INTEGER data type. I have also tried it on TIMESTAMP and REAL data types. I'm using a local sqlite database

**Reply by wangxiaoying** [reply]
That's weird, we have a [CI test](https://github.com/sfu-db/connector-x/blob/main/connectorx-python/connectorx/tests/test_sqlite.py#L158) which partition on an integer column and it passes, the DDL can be found [here](https://github.com/sfu-db/connector-x/blob/main/scripts/sqlite.sql#L3). Can you confirm that the error remains in the [newest version](https://pypi.org/project/connectorx/0.3.1a8/)? If it is, can you provide a toy example (including DDL, inserting sample data and the query) that throws the error so we can reproduce the error?
